src/reconstruction/refinement/addReactionGEM.m

```diff
@@ -69,6 +69,8 @@
 end
 
 nMet = length(model.mets)+1;
+orignMets = numel(model.mets);

```

Comment:
length/numel is the same for a vector. and admittedly length is preferable. Its just an old habit of mine to use numel, as it does not hide the other dimensions. length gives you the largest dimension (essentially max(size(x))) while numel gives yout the multiplication (essentially prod(size(x))).
As for orignMets:
You could add multiple metabolites in one call of addReactionGEM (having multiple metabolites in the reaction not yet in the model), so while yes, there is a direct link between orignMets and nMet (namely the number of added metabolites), this would require an additional call. 
As mentioned, nMet is modified later on to know the current number of metabolites in the model, and we need to update the fields for all of those. And its faster if we only do this once and not multiple times.